clang/lib/DPCT/ASTTraversal.cpp

```diff
@@ -14744,3 +14744,58 @@ void CudaUuidRule::runRule(
 }
 
 REGISTER_RULE(CudaUuidRule, PassKind::PK_Analysis)
+
+void ForLoopUnrollRule::registerMatcher(ast_matchers::MatchFinder &MF) {
+  MF.addMatcher(forStmt().bind("loop"), this);
+}
+
+void ForLoopUnrollRule::runRule(
+    const ast_matchers::MatchFinder::MatchResult &Result) {
+  if (!DpctGlobalInfo::isOptimizeMigration()) {
+    return;
```

Comment:
can move the check into like L14749.   Also consider to add the rule into separate file.